build-tools/packages/build-tools/src/test/biomeConfig.test.ts

```diff
@@ -0,0 +1,214 @@
+			});
+
+			it("formatter.include has correct value", async () => {
+				const testFile = path.resolve(testDataPath, "biome/pkg-a/biome.jsonc");
+				const actual = await loadBiomeConfig(testFile);
+
+				assert(actual.formatter?.include?.includes("include-formatter/**"));
+				assert(actual.formatter?.include?.length === 1);
+			});
+
+			it("formatter.ignore has correct value", async () => {
+				const testFile = path.resolve(testDataPath, "biome/pkg-a/biome.jsonc");
+				const actual = await loadBiomeConfig(testFile);
+
+				assert(actual.formatter?.ignore?.includes("ignore-formatter/**"));
+				assert(actual.formatter?.ignore?.length === 1);
+			});
+
+			it("linter.include has correct value", async () => {
+				const testFile = path.resolve(testDataPath, "biome/pkg-a/biome.jsonc");
+				const actual = await loadBiomeConfig(testFile);
+
+				assert(actual.linter?.include?.includes("include-linter/**"));
+				assert(actual.linter?.include?.length === 1);
+			});
+
+			it("linter.ignore has correct value", async () => {
+				const testFile = path.resolve(testDataPath, "biome/pkg-a/biome.jsonc");
+				const actual = await loadBiomeConfig(testFile);
+
+				assert(actual.linter?.ignore?.includes("ignore-linter/**"));
+				assert(actual.linter?.ignore?.length === 1);
+			});
+		});
+	});
+
+	describe("getSettingValuesFromBiomeConfig", async () => {
+		describe("extends from a single config", async () => {
+			let testConfig: BiomeConfig;
+			before(async () => {
+				const testFile = path.resolve(testDataPath, "biome/pkg-a/biome.jsonc");
+				testConfig = await loadBiomeConfig(testFile);
+			});
+
+			it("formatter ignore settings are merged with root", async () => {
+				const ignores = await getSettingValuesFromBiomeConfig(
+					testConfig,
+					"formatter",
+					"ignore",
+				);
+				assert(ignores.has("pkg-a-ignore/**"));
+				assert(ignores.has("ignore-formatter/**"));
+				assert(ignores.size === 2);
+			});
+
+			it("linter ignore settings are merged with root", async () => {
+				const ignores = await getSettingValuesFromBiomeConfig(testConfig, "linter", "ignore");
+				assert(ignores.has("pkg-a-ignore/**"));
+				assert(ignores.has("ignore-linter/**"));
+				assert(ignores.size === 2);
+			});
+
+			it("formatter include settings are merged with root", async () => {
+				const includes = await getSettingValuesFromBiomeConfig(
+					testConfig,
+					"formatter",
+					"include",
+				);
+				assert(includes.has("pkg-a-include/**"));
+				assert(includes.has("include-formatter/**"));
+				assert(includes.size === 2);
+			});
+
+			it("linter include settings are merged with root", async () => {
+				const includes = await getSettingValuesFromBiomeConfig(
+					testConfig,
+					"linter",
+					"include",
+				);
+				assert(includes.has("pkg-a-include/**"));
+				assert(includes.has("include-linter/**"));
+				assert(includes.size === 2);
+			});
+		});
+	});
+
+	describe("getBiomeFormattedFiles", async () => {
```

Comment:
Should there be some files in the test set that Biome would always ignore because of extension?

Reply:
As noted in another comment thread, there's no "always ignore/always include" list of extensions. But this made me realize we could add some test cases to check for extension-level includes/ignores, not just path or filename. E.g. always exclude "*.md" and always include "*.ts".